pkg/util/mount/safe_format_and_mount_test.go

```diff
@@ -159,30 +159,20 @@ func TestSafeFormatAndMount(t *testing.T) {
 			mountErrs:   []error{fmt.Errorf("unknown filesystem type '(null)'"), nil},
 			execScripts: []ExecArgs{
 				{"fsck", []string{"-a", "/dev/foo"}, "", nil},
-				{"lsblk", []string{"-n", "-o", "FSTYPE", "/dev/foo"}, "\n", nil},
+				{"blkid", []string{"-p", "-s", "TYPE", "-o", "value", "/dev/foo"}, "", nil},
 				{"mkfs.xfs", []string{"/dev/foo"}, "", nil},
 			},
 			expectedError: nil,
 		},
 		{
-			description: "Test that 'lsblk' is called and reports ext4 partition",
+			description: "Test that 'blkid' is called and reports ext4 partition",
 			fstype:      "ext3",
 			mountErrs:   []error{fmt.Errorf("unknown filesystem type '(null)'")},
 			execScripts: []ExecArgs{
 				{"fsck", []string{"-a", "/dev/foo"}, "", nil},
-				{"lsblk", []string{"-n", "-o", "FSTYPE", "/dev/foo"}, "\next4\n", nil},
+				{"blkid", []string{"-p", "-s", "TYPE", "-o", "value", "/dev/foo"}, "LVM2_member\n", nil},
 			},
-			expectedError: fmt.Errorf("failed to mount the volume as \"ext3\", it already contains unknown data, probably partitions. Mount error: unknown filesystem type '(null)'"),
-		},
-		{
-			description: "Test that 'lsblk' is called and reports empty partition",
```

Comment:
So, what blkid returns when a device contains a partition? lsblk detected it and kubelet did not reformat the device. Now it reformats it, wiping all user data.

Reply:
@jsafrane 

In addition to print  FS type (by `-s TYPE`), we can print `PTTYPE` too.

`blkid -p -s TYPE -s PTTYPE -o export <device>`:

Example output if device contains parititions:

```
# blkid -p -s TYPE -s PTTYPE -o export /dev/sda1
DEVNAME=/dev/sda
PTTYPE=dos
```

Example output if device does not contains partitions:

```
# blkid -p -s TYPE -s PTTYPE -o export /dev/sda1
DEVNAME=/dev/sda1
TYPE=ext2
```

PTTYPE is partition type, referring to docs: https://www.kernel.org/pub/linux/utils/util-linux/v2.21/libblkid-docs/libblkid-Partitions-probing.html

Reply:
@jsafrane Updated. I removed the last "empty partitions" test case, because we only need to know if disk has partitions or not. Though it's possible to get partition lists from [`PART_ENTRY_*` tags](https://www.kernel.org/pub/linux/utils/util-linux/v2.21/libblkid-docs/libblkid-Partitions-probing.html).